src/pe/utils.rs

```diff
@@ -64,6 +64,8 @@ fn section_read_size(section: &section_table::SectionTable, file_alignment: u32)
 
     if virtual_size == 0 {
         read_size
+    } else if size_of_raw_data == 0 {
```

Comment:
i'm a little worried about this; this function is used all over goblin, can you add a comment why this check is done, and why it returns the virtual_size in this case? also there are no tests added to verify anything here

Reply:
analogically to `virtual_size == 0` -> used `read_size`, if `read_size == 0` -> used `virtual_size`. i saw [pefile](https://github.com/erocarrera/pefile/tree/master) does this and there are no errors in parsing the AMD sample, so i added this check too and it works

Reply:
also added test + sample for this case: took file from https://www.amd.com/en/resources/support-articles/faqs/GPU-601.html + packed with upx.

is it ok if i add this file to tests?

Reply:
I have no idea what the license of that binary is :sweat_smile:  so probably let's not add it for now; also, it seems to parse just fine for me without this patch? e.g.:

```
cargo run --example rdr -- ~/Downloads/amdcleanuputility.exe
```
has no issue, neither does `bingrep`

Reply:
Parse `amdcleanuputility.exe` itself is fine. But if binary will packed with upx, parsing goes broken, while binary can be run properly.

Ok, test removed from PR